autoconfigure/storage-elasticsearch-aws/src/main/java/zipkin/autoconfigure/storage/elasticsearch/aws/ElasticsearchDomainEndpoint.java

```diff
@@ -13,73 +13,75 @@
 
-  @Override
-  public List<String> get() {
-    try (Response response = client.newCall(describeElasticsearchDomain).execute()) {
-      String body = response.body().string();
-      if (!response.isSuccessful()) {
-        String message =
-            describeElasticsearchDomain.url().encodedPath()
-                + " failed with status "
-                + response.code();
-        if (!body.isEmpty()) message += ": " + body;
-        throw new IllegalStateException(message);
-      }
-      JsonReader endpointReader = JsonReader.of(new Buffer().writeUtf8(body));
-      endpointReader = enterPath(endpointReader, "DomainStatus", "Endpoints");
-      if (endpointReader != null) endpointReader = enterPath(endpointReader, "vpc");
-
-      if (endpointReader == null) {
-        endpointReader =
-            enterPath(JsonReader.of(new Buffer().writeUtf8(body)), "DomainStatus", "Endpoint");
-      }
+  @Override public List<String> get() {
+    HttpStatus status;
+    String body;
+    try {
+      AggregatedHttpResponse res = client.execute(describeElasticsearchDomain).aggregate().join();
+      status = res.status();
+      // As the domain endpoint is read only once per startup. We don't worry about pooling etc.
+      // This allows for easier debugging and less try/finally state management.
+      body = res.contentUtf8();
+    } catch (RuntimeException | Error e) {
+      throw new RuntimeException("couldn't lookup AWS ES domain endpoint",
+          e instanceof CompletionException ? e.getCause() : e
+      );
+    }
 
-      if (endpointReader == null) {
-        throw new IllegalStateException(
-            "Neither DomainStatus.Endpoints.vpc nor DomainStatus.Endpoint were present in response: "
-                + body);
-      }
+    if (!status.codeClass().equals(HttpStatusClass.SUCCESS)) {
+      String message = describeElasticsearchDomain.path() + " failed with status " + status;
+      if (!body.isEmpty()) message += ": " + body;
+      throw new RuntimeException(message);
+    }
 
-      String endpoint = endpointReader.nextString();
-      if (!endpoint.startsWith("https://")) {
-        endpoint = "https://" + endpoint;
-      }
-      log.fine("using endpoint " + endpoint);
-      return Collections.singletonList(endpoint);
+    String endpoint;
+    try {
+      JsonNode root = JSON.readTree(body);
+      endpoint = root.at("/DomainStatus/Endpoints/vpc").textValue();
+      if (endpoint == null) endpoint = root.at("/DomainStatus/Endpoint").textValue();
     } catch (IOException e) {
-      throw new IllegalStateException("couldn't lookup domain endpoint", e);
+      throw new AssertionError("Unexpected to have IOException reading a string", e);
     }
+
+    if (endpoint == null) {
+      throw new RuntimeException(
+          "Neither DomainStatus.Endpoints.vpc nor DomainStatus.Endpoint were present in response: "
+              + body);
+    }
+
+    if (!endpoint.startsWith("https://")) endpoint = "https://" + endpoint;
```

Comment:
This is the same code as previously, so probably ok, but just noticed that maybe it should also check for `http://`

Reply:
the check existing at all is incorrect, but I'll add a comment saying why it is.